pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -35,14 +39,20 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 	existingRs := existingVersionedResources(d.existingRs)
 	existingRsGrouped := d.groupResources(existingRs.Versioned)
 
+	existingNonVerRsGrouped := d.groupNonVerResources(existingRs.NonVersioned)
+	err := d.checkForMaxDurationAnn(existingRsGrouped, existingNonVerRsGrouped)
```

Comment:
`checkForMaxDurationAnn`: Hmm, I think we are doing a lot for then just checking for the max-duration annotation. I can't think of a name right now that would describe the actual functionality over here, but let's try to see if we can find something better.

Reply:
How about -> checkAndUpdateResWithMaxDurationAnn ?